internal/build/buildconfig/maker.go

```diff
@@ -33,12 +39,28 @@ func NewMaker(helper build.Helper, scheme *runtime.Scheme) Maker {
 	}
 }
 
-func (m *maker) MakeBuildConfigTemplate(mod kmmv1beta1.Module, mapping kmmv1beta1.KernelMapping, targetKernel, containerImage string, pushImage bool) (*buildv1.BuildConfig, error) {
+func (m *maker) MakeBuildConfigTemplate(mod kmmv1beta1.Module, mapping kmmv1beta1.KernelMapping, targetKernel, containerImage string,
+	pushImage bool, syncronizedKernelOsDtkMapping syncronizedmap.SyncronizedKernelOsDtkMapping) (*buildv1.BuildConfig, error) {
+
 	kmmBuild := m.helper.GetRelevantBuild(mod, mapping)
 
+	dtkImage, ok := syncronizedKernelOsDtkMapping.GetImage(targetKernel)
+	if !ok {
+		return nil, fmt.Errorf("could not get DTK image for kernel %v", targetKernel)
+	}
+
+	overrides := []kmmv1beta1.BuildArg{
+		{
+			Name:  "KERNEL_VERSION",
+			Value: targetKernel,
+		},
+	}
+	if dockerfile := kmmBuild.Dockerfile; strings.Contains(dockerfile, dtkBuildArg) {
```

Comment:
```suggestion
	if strings.Contains(dockerfile, dtkBuildArg) {
```

Reply:
I am not sure what you meant here. `dockerfile` isn't defined if you remove the variable definition.

Reply:
sorry, bug in suggestion:

if strings.Contains(kmmBuild.dockerfile, dtkBuildArg) {

Reply:
You are right. I did it cause I used `dockerfile` in the loop but I changed it and forgot to remove it from the `if` statement as well.